pkg/internal/infrastructure/templates/main.tpl.tf

```diff
@@ -121,7 +132,33 @@ resource "google_compute_subnetwork" "subnetwork-internal" {
     delete = "5m"
   }
 }
-{{- end}}
+{{- end }}
+
+{{ if .networks.privateServiceConnect }}
+  resource "google_compute_global_address" "default" {
+  provider     = google-beta
+  name         = "{{ .clusterName }}"
+  address_type = "INTERNAL"
+  purpose      = "PRIVATE_SERVICE_CONNECT"
+  network      = {{ .vpc.name }}
+  address      = "{{ .networks.privateServiceConnect.address }}"
+}
+
+// There is an issue currently that TF will always delete and recreate the address during infrastructure reconciliation
+// because it cannot handle the labels properly.
```

Comment:
Yes it is recreated in every infra reconciliation. The reason is that between the terraform and the GCP API the labels value is oscillating between an empty map and a nil value. This makes TF believe that the resource is changed and it recreates it which may interrupt open connections.

I tried all the combinations  e.g. specifying the labels as empty map, not specifying them at all and none of them stopped this behavior. Also it is not possible to initialise the map with any kind of value because it is prohibited by the GCP API for endpoints used in PSC